Reduce use of mainFrameJSContext in WebKitTestRunner

achristensen07 opened the pull request:
#### b05ec723eb45d12d5df692990551d89cce4dc919
<pre>
Reduce use of mainFrameJSContext in WebKitTestRunner
<a href="https://bugs.webkit.org/show_bug.cgi?id=273246">https://bugs.webkit.org/show_bug.cgi?id=273246</a>

Reviewed by Sihui Liu and Charlie Wolfe.

With site isolation the main frame might be in another process.

* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageGetApplicationManifest):
(WKPageGetApplicationManifest_b): Deleted.
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
(WTR::InjectedBundle::postAddChromeInputField): Deleted.
(WTR::InjectedBundle::postSetTextInChromeInputField): Deleted.
(WTR::InjectedBundle::postSelectChromeInputField): Deleted.
(WTR::InjectedBundle::postGetSelectedTextInChromeInputField): Deleted.
(WTR::InjectedBundle::postFocusWebView): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::addChromeInputField):
(WTR::TestRunner::setTextInChromeInputField):
(WTR::TestRunner::selectChromeInputField):
(WTR::TestRunner::getSelectedTextInChromeInputField):
(WTR::TestRunner::focusWebView):
(WTR::TestRunner::loadedSubresourceDomains):
(WTR::TestRunner::setOpenPanelFilesMediaIcon):
(WTR::TestRunner::getApplicationManifestThen):
(WTR::TestRunner::callAddChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callSetTextInChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callSelectChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callGetSelectedTextInChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callFocusWebViewCallback): Deleted.
(WTR::makeDomainsValue): Deleted.
(WTR::TestRunner::callDidReceiveLoadedSubresourceDomainsCallback): Deleted.
(WTR::TestRunner::didGetApplicationManifest): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::adoptAndCallCompletionHandler):
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::TestController::loadedSubresourceDomains):
(WTR::LoadedSubresourceDomainsCallbackContext::LoadedSubresourceDomainsCallbackContext): Deleted.
(WTR::loadedSubresourceDomainsCallback): Deleted.
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveLoadedSubresourceDomains): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::loadedSubresourceDomains): Deleted.
(WTR::TestController::clearLoadedSubresourceDomains): Deleted.

Canonical link: <a href="https://commits.webkit.org/277990@main">https://commits.webkit.org/277990@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/afcac613cae06394d1e44c476a3662d0113844fd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49141 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28418 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52154 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51939 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/45236 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51445 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34404 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25995 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40164 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/50795 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25982 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42360 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21277 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23438 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/43534 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7465 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45366 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44041 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53848 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24219 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20425 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47482 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25497 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42565 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46466 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26313 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/7045 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25216 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
```diff
@@ -512,26 +512,6 @@ void initializeWebViewConfiguration(const char* libraryPath, WKStringRef injecte
-    }];
-}
-
-void TestController::clearLoadedSubresourceDomains()
```
I don't understand why it's okay to remove these.

achristensen07 replied:
we're using the platform-independent versions now

szewai commented on Tools/WebKitTestRunner/TestController.cpp:
```diff
@@ -1952,47 +1952,53 @@ void TestController::didReceiveMessageFromInjectedBundle(WKStringRef messageName
     if (WKStringIsEqualToUTF8CString(messageName, "SetStatisticsPrevalentResourceForDebugMode")) {
         WKStringRef hostName = stringValue(messageBody);
-        TestController::singleton().setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));
+        setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));
```
You could do `return setStatisticsPrevalentResourceForDebugMode(hostName, WTFMove(completionHandler));` as the other places.

achristensen07 replied:
I could.  I typically do that only if it makes the difference between having { } and not having { }, but it's a stylistic thing that doesn't matter much.

achristensen07 committed:
Reduce use of mainFrameJSContext in WebKitTestRunner
https://bugs.webkit.org/show_bug.cgi?id=273246

Reviewed by Sihui Liu and Charlie Wolfe.

With site isolation the main frame might be in another process.

* Source/WebKit/UIProcess/API/C/WKPage.cpp:
(WKPageGetApplicationManifest):
(WKPageGetApplicationManifest_b): Deleted.
* Source/WebKit/UIProcess/API/C/WKPagePrivate.h:
* Tools/WebKitTestRunner/InjectedBundle/Bindings/TestRunner.idl:
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.cpp:
(WTR::InjectedBundle::didReceiveMessageToPage):
(WTR::InjectedBundle::postAddChromeInputField): Deleted.
(WTR::InjectedBundle::postSetTextInChromeInputField): Deleted.
(WTR::InjectedBundle::postSelectChromeInputField): Deleted.
(WTR::InjectedBundle::postGetSelectedTextInChromeInputField): Deleted.
(WTR::InjectedBundle::postFocusWebView): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/InjectedBundle.h:
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.cpp:
(WTR::TestRunner::addChromeInputField):
(WTR::TestRunner::setTextInChromeInputField):
(WTR::TestRunner::selectChromeInputField):
(WTR::TestRunner::getSelectedTextInChromeInputField):
(WTR::TestRunner::focusWebView):
(WTR::TestRunner::loadedSubresourceDomains):
(WTR::TestRunner::setOpenPanelFilesMediaIcon):
(WTR::TestRunner::getApplicationManifestThen):
(WTR::TestRunner::callAddChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callSetTextInChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callSelectChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callGetSelectedTextInChromeInputFieldCallback): Deleted.
(WTR::TestRunner::callFocusWebViewCallback): Deleted.
(WTR::makeDomainsValue): Deleted.
(WTR::TestRunner::callDidReceiveLoadedSubresourceDomainsCallback): Deleted.
(WTR::TestRunner::didGetApplicationManifest): Deleted.
* Tools/WebKitTestRunner/InjectedBundle/TestRunner.h:
* Tools/WebKitTestRunner/TestController.cpp:
(WTR::adoptAndCallCompletionHandler):
(WTR::TestController::didReceiveAsyncMessageFromInjectedBundle):
(WTR::TestController::loadedSubresourceDomains):
(WTR::LoadedSubresourceDomainsCallbackContext::LoadedSubresourceDomainsCallbackContext): Deleted.
(WTR::loadedSubresourceDomainsCallback): Deleted.
* Tools/WebKitTestRunner/TestController.h:
* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::didReceiveMessageFromInjectedBundle):
(WTR::TestInvocation::didReceiveLoadedSubresourceDomains): Deleted.
* Tools/WebKitTestRunner/TestInvocation.h:
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::loadedSubresourceDomains): Deleted.
(WTR::TestController::clearLoadedSubresourceDomains): Deleted.

Canonical link: https://commits.webkit.org/277990@main
